src/util/codec/mysql/json/functions.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// FIXME: remove following later
+#![allow(dead_code)]
+
+use std::{u32, char};
+use super::super::Result;
+use super::json::Json;
+
+const ESCAPED_UNICODE_BYTES_SIZE: usize = 4;
+
+impl Json {
+    pub fn unquote(&self) -> Result<String> {
+        match *self {
+            Json::String(ref s) => unquote_string(s),
+            _ => Ok(format!("{:?}", self)),
+        }
+    }
+}
+
+// unquote_string recognizes the escape sequences shown in:
+// https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#
+// json-unquote-character-escape-sequences
+pub fn unquote_string(s: &str) -> Result<String> {
+    let mut ret = String::with_capacity(s.len());
```

Comment:
In tidb I use a byte buffer to store utf8 encoded bytes. Call `String()` on that we can get a correct string, because go default string encoding format is utf8.